shelley/chain-and-ledger/formal-spec/utxo.tex

```diff
@@ -580,7 +580,14 @@ \subsection{Witnesses}
       \right\}
   \end{align*}
 
-    \begin{align*}
+  \begin{align*}
+    & \hspace{-0.8cm}\fun{certWitsNeeded} \Tx \to \powerset{\Credential}
+    & \text{certificates with witnesses} \\
+    &  \hspace{-0.8cm}\fun{certWitsNeeded}~\var{tx} = \\
+    & \cup\{\cwitness{c} \mid c \in \txcerts{tx} \setminus (\DCertRegKey\cup\DCertMir)\}
```

Comment:
```suggestion
    & \{\cwitness{c} \mid c \in \txcerts{tx} \setminus (\DCertRegKey\cup\DCertMir)\}
```

Reply:
`cwitness` returns a set of credentials, since the stake pool registration certs can need a more than one. so without the union, we have a set of sets, and we need to flatten it. yea?

Reply:
ok, I see, but then maybe better use an "uppercase" union ?